### PR TITLE
Initial java snake

### PR DESCRIPTION
**Pull request by codeallthethingz**
Added tests and boiler plate.  Uses Spark as the HTTP server framework.

Theoretically this will deploy to heroku but that's what I'll work on next.

Now works on heroku.

### CODE REVIEW

File: src/main/java/io/battlesnake/starter/Snake.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+     *
+     * @param args are ignored.
+     */
+    public static void main(String[] args) {
+        String port = System.getProperty("PORT");
+        if (port != null) {
+            LOG.info("Found system provided port: {}", port);
+        } else {
+            port = "8080";
+        }
```
Comment (joram) [inline src/main/java/io/battlesnake/starter/Snake.java]: should we log here instead, so we always know which port it's using?

Comment (codeallthethingz) [reply src/main/java/io/battlesnake/starter/Snake.java]: :+1:

**Comment by torbensky** [comment]
This is overall solid, but I'm not a fan of the Map stuff. I'd prefer either a `JsonNode` or a POJO to interact with. If we have a swagger spec, we could easily generate POJOs for the corresponding API data types. If that isn't desirable, a JsonNode instance would provide a nicer API to work with than the Map. Thoughts?

**Comment by codeallthethingz** [comment]
Yeah, I went back and forth on that, and ended up on map because it takes a lot less explaining. 

@joram your call, JsonNode has a bunch of helper functions for casting to ints and strings automatically, as well as path methods that find an element and return an empty object rather than null.

**Comment by torbensky** [comment]
That would a different standard than the Go snake, for example, which has DTO objects available. Also the node snake will have a proper JSON object as well. The Map structure as it stands now would be needlessly difficult to work with imo because it flattens JSON into a string value. For example, `board.food` and `board.snakes` would just be a string value.

If we do want this to be simpler, we probably should change the Go stuff then: https://github.com/battlesnakeio/starter-snake-go/blob/master/api.go

**Comment by codeallthethingz** [comment]
@torbensky ObjectMapper doesn't flatten the structure.  You end up with a map of maps and lists.

**Comment by torbensky** [comment]
@codeallthethingz Ahh yes, that was the response body that was Map<string,string>, good point. I do think it was a good call to change to a JsonNode still. Just a bit easier to work with, but not opinionated about a particular model.

👍

**Pull request merged**